DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_port_state.py

```diff
@@ -0,0 +1,102 @@
+import pytest, time
+from dent_os_testbed.lib.ip.ip_link import IpLink
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen
+)
+
+from datetime import datetime
+
+pytestmark = [
+    pytest.mark.suite_functional_l1,
+    pytest.mark.asyncio,
+    pytest.mark.usefixtures("cleanup_tgen")
+]
+
+async def port_state(testbed, counter, software_reboot = False):
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    ports = tgen_dev.links_dict[device_host_name][1]
+    timeout = 30
+
+    for _ in range(counter):
+        start_time = datetime.now()
+        links_present = []
+        for port in ports:
+            rc, out = await dent_dev.run_cmd(f"ifconfig -a | grep -w '{port}' | wc -l")
```

Comment:
ifconfig -a print all ports. Please do not call it multiple times. Do a single API call and parse the output.